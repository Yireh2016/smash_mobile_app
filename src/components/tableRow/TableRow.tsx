import React from 'react';
import { StyleSheet, View } from 'react-native';

export interface Column {
  component: Element;
  align: string;
  width?: number;
}

interface CustomColumnsTableProps {
  columns: Column[];
}

const CustomColumnsTable: React.FunctionComponent<CustomColumnsTableProps> = ({
  columns,
}) => {
  const styles = StyleSheet.create({
    threeColumnsLayout: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
    },
    alignRight: {
      alignItems: 'flex-end',
    },
    alignCenter: {
      alignItems: 'center',
    },
    alignLeft: {
      alignItems: 'flex-start',
    },
  });

  const columnsMap = columns.map((column: Column, i: number) => {
    const { component, align, width } = column;
    let columnAlign;

    switch (align) {
      case 'left':
        columnAlign = styles.alignLeft;
        break;

      case 'center':
        columnAlign = styles.alignCenter;
        break;

      case 'right':
        columnAlign = styles.alignRight;
        break;

      default:
        columnAlign = styles.alignLeft;
        break;
    }

    return (
      <View
        key={i}
        style={{
          ...columnAlign,
          width: width ? `${width}%` : `${100 / columns.length}%`,
        }}
      >
        {component}
      </View>
    );
  });

  return <View style={styles.threeColumnsLayout}>{columnsMap}</View>;
};

export default CustomColumnsTable;
