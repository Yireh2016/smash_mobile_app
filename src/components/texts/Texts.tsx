/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { Text } from 'react-native';
import theme from '../../theme/theme';
import { ChildrenProp } from '../../types/common/commonTypes';

interface TextsProps extends ChildrenProp {
  rest?: any;
  style?: {};
  color?: string;
  isBold?: boolean;
  onPress?(): void;
}

const openSans = {
  bold: { fontFamily: theme.font.bold },
  regular: { fontFamily: theme.font.regular },
};

export const MainTitle: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  ...rest
}) => {
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...openSans.bold,
        fontSize: 28,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const H1: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  ...rest
}) => {
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...openSans.bold,
        fontSize: 24,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const H2: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  ...rest
}) => {
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...openSans.bold,
        fontSize: 20,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const H3: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  ...rest
}) => {
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...openSans.bold,
        fontSize: 18,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const MainParagraph: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  onPress,
  color = theme.palette.primaryContrast,
  isBold = false,
  ...rest
}) => {
  const fontFamily = isBold ? openSans.bold : openSans.regular;
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...fontFamily,
        fontSize: 18,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const P1: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  onPress,
  color = theme.palette.primaryContrast,
  isBold = false,
  ...rest
}) => {
  const fontFamily = isBold ? openSans.bold : openSans.regular;
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...fontFamily,
        fontSize: 16,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const P2: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  isBold = false,
  ...rest
}) => {
  const fontFamily = isBold ? openSans.bold : openSans.regular;
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...fontFamily,
        fontSize: 14,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const P3: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.primaryContrast,
  onPress,
  isBold = false,
  ...rest
}) => {
  const fontFamily = isBold ? openSans.bold : openSans.regular;
  return (
    <Text
      {...rest}
      onPress={onPress}
      style={{
        ...style,
        ...fontFamily,
        fontSize: 12,
        color: color,
      }}
    >
      {children}
    </Text>
  );
};

export const SmashLink: React.FunctionComponent<TextsProps> = ({
  children,
  style,
  color = theme.palette.secondary,
  onPress,
  ...rest
}) => {
  return (
    <P1
      {...rest}
      color={color}
      isBold={true}
      style={{ ...style }}
      onPress={onPress}
    >
      {children}
    </P1>
  );
};
