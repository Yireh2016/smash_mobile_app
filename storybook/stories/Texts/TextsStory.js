import React from 'react';

import { storiesOf } from '@storybook/react-native';

import CenterLayout from '../../../src/layouts/centerLayout/CenterLayout';

import { H1, H2, P1, P2 } from '../../../src/components/texts/Texts';

export default () =>
  storiesOf('Texts', module)
    .addDecorator((getStory) => (
      <CenterLayout color="#fff">{getStory()}</CenterLayout>
    ))
    .add('Texts', () => {
      return (
        <>
          <H1>Important</H1>
          <H2>Less Important</H2>
          <P1>normal</P1>
          <P2>I did not exist</P2>
        </>
      );
    });
