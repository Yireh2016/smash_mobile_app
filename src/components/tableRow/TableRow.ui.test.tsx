import * as React from 'react';
import renderer from 'react-test-renderer';
import { P1, H3 } from '../texts/Texts';
import TableRow from './TableRow';

it('TableRow primary renders correctly', () => {
  const ComponentProps = {
    columns: [
      {
        component: <P1>Est. pay off time</P1>,
        align: 'left',
      },
      {
        component: <H3>test</H3>,
        align: 'right',
      },
    ],
  };
  const Element = () => {
    return <TableRow {...ComponentProps} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
