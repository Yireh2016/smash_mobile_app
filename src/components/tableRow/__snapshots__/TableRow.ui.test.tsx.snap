// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow primary renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
    >
      Est. pay off time
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 18,
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
